### DISCUSSION TITLE
Uvicorn 0.18.x --reload do not work

### DISCUSSION THREAD

**Discussion by KStenK** [Kludex/uvicorn#1586] [Potential Issue] [answered]
We are using **Uvicorn[standard]** with the **FastApi** framework.
For IDE, we are using **Pycharm 2022.2**, and the python interpreter is in **docker-compose**.

I see that in the uvicorn 0.18.0 version, the **Watchdog** has been swapped out with **Watchfiles**.

After upgrading uvicorn from 0.17.6 to 0.18.2 or to any 0.18.x version, I see that the `--reload` option no longer works for us.

When I start the Fastapi, the `Will watch for changes in these directories` is pointing to the correct directory, and if I check files manually for updates, then they are updating too, but **Watchfiles** do not catch those updates.

Any clues why it is not working anymore?

**Comment by Kludex** [comment]
@samuelcolvin 👀

**Reply by samuelcolvin** [reply]
First of all, uvicorn before v0.18 used "watchgod" not "watchdog" the name confusion was the main reason I changed the name to "watchfiles" during the rewrite.

There are a number of issues with docker and file system change notifications, particularly related to:
* `--platform linux/amd64` / QEMU (see https://github.com/samuelcolvin/watchfiles/issues/167), this is fixed in v0.16.1 of watchfiles, run `pip install -U watchfiles` to fix
* [WSL](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux) (see https://github.com/samuelcolvin/watchfiles/issues/169), this can't be fixed completely in watchfiles AFAIK, but can be fixed by setting the environment variable `WATCHFILES_FORCE_POLLING`, again you'll need the latest watchfiles installed with `pip install -U watchfiles`

Hope that helps.

**Comment by KStenK** [comment, marked as answer]
Hi @samuelcolvin,

Sorry, yes, Watchgod, not Watchdog 😁
 
After setting the `WATCHFILES_FORCE_POLLING` env variable to `True`, the  `--reload ` started to work.

Thank you.
